### PR TITLE
add oauth2 DSL

### PR DESCRIPTION
**Pull request by masarakki**

### CODE REVIEW

File: lib/wine_bouncer/extension.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module WineBouncer
+  module Extension
+    def oauth2(*scopes)
+      scopes = Array(scopes)
+      description = if respond_to?(:route_seting) # >= grape-0.10.0 
+        route_setting(:description)
+      else
+        @last_description ||= {}
+      end
+      # case WineBouncer.configuration.auth_strategy
```
Comment (masarakki) [inline lib/wine_bouncer/extension.rb]: I can't write spec with this branching because `configuration` is required before calling `oauth2` method.

**Comment by antek-drzewiecki** [comment]
Nice! I am merging this and adding grape 10 to the build matrix.

**Comment by antek-drzewiecki** [comment]
Thanks!

**Comment by antek-drzewiecki** [comment]
@masarakki i am thinking to change the dsl from oauth2(*scopes) to authorize!(*scopes).
The naming suggest less of the implementation that is beneath the authentication of your DSL but more of the functionality. 
What do you think of it?

**Comment by masarakki** [comment]
I cared that `authenticate` may conflict with other libraries (ex grape himself),
so I named `oauth2`. (I think no one uses multiple oauth2 authenticate libraries at once.)

and `authorize!(*scopes)` is few information to understand how authorize because there are a lot of authorize methods like `Basic`, `Digest`. `OAuth`, `OAuth2`, ...

**Comment by antek-drzewiecki** [comment]
You are right. It might result in conflicts. oauth2 will do.

**Pull request merged**